packages/circuit-ui/components/Button/Button.tsx

```diff
@@ -157,34 +157,93 @@ const Content = styled.span<{ isLoading: boolean }>(
   contentLoadingStyles,
 );
 
-const COLOR_MAP = {
+const PRIMARY_COLOR_MAP = {
```

Comment:
Now that we're using CSS variables, I'd refactor this code to move away from this object and replace it with a `switch` statement and explicit styles. Would help readability I think.

Reply:
Good idea! Just replaced the tokens for now, but you're right that this is a good opportunity to make this a bit clearer. Will address (in January)

Reply:
Was contemplating this again and realized it's actually a matrix between `variant` and `destructive`. Would keep it for now and refactor with the new button design.